analyzer/codechecker_analyzer/cmd/checkers.py

```diff
@@ -179,240 +208,409 @@ def add_arguments_to_parser(parser):
-            analyzer_class = analyzer_types.supported_analyzers[analyzer]
-
-            configs = analyzer_class.get_checker_config(config_handler,
-                                                        analyzer_environment)
-            if not configs:
-                analyzer_failures.append(analyzer)
-                continue
-
-            rows.extend((':'.join((analyzer, c[0])), c[1]) if 'details' in args
-                        else (':'.join((analyzer, c[0])),) for c in configs)
-
-        if rows:
-            print(twodim.to_str(args.output_format, header, rows))
-
-        if analyzer_failures:
-            LOG.error("Failed to get checker configuration options for '%s' "
-                      "analyzer(s)! Please try to upgrade your analyzer "
-                      "version to use this feature.",
-                      ', '.join(analyzer_failures))
-            sys.exit(1)
 
-        return
+def __format_guideline(guideline: str) -> str:
+    """
+    Convert guideline rules to human-readable format.
+    guideline -- Dictionary in the following format:
+                    {"guideline_1": ["rule_1", "rule_2"]}
+    """
+    return ' '.join('Related {} rules: {}'.format(g, ', '.join(r))
+                    for g, r in guideline.items())
 
-    if args.guideline is not None and len(args.guideline) == 0:
-        result = defaultdict(set)
 
-        for _, guidelines in context.guideline_map.items():
-            for guideline, rules in guidelines.items():
-                result[guideline] |= set(rules)
+def __format_row(row: Tuple, format: str) -> Tuple:
+    """
+    Perform some formatting of the detailed checker info.
+    row -- A tuple with detailed checker info coming from
+           __get_detailed_checker_info() function.
+    format -- Output format given by -o flag.
+    """
+    if format == 'json':
+        state = row[0] == CheckerState.enabled
+        guideline = row[4]
+    else:
+        state = '+' if row[0] == CheckerState.enabled else \
+            '-' if row[0] == CheckerState.disabled else \
+            '?'
+        guideline = __format_guideline(row[4])
+
+    return (state, row[1], row[2], row[3], guideline, row[5])
 
-        header = ['Guideline', 'Rules']
-        if args.output_format in ['csv', 'json']:
-            header = list(map(uglify, header))
 
-        if args.output_format == 'json':
-            rows = [(g, sorted(list(r))) for g, r in result.items()]
+def __print_checkers_custom_format(checkers, cl: CheckerLabels):
+    for checker in checkers:
+        if checker[0] == CheckerState.enabled:
+            status = '+'
+        elif checker[0] == CheckerState.disabled:
+            status = '-'
         else:
-            rows = [(g, ', '.join(sorted(r))) for g, r in result.items()]
+            status = '?'
+
+        labels = cl.labels_of_checker(checker[1], checker[2])
+
+        print(checker[1])
+        print('  Enabled:', status)
```

Comment:
Ok, I used label `Status` here, however, in JSON format I still used `enabled` because its value is boolean. `enabeld: true` seems more intuitive than `status: true`. I think this inconsistency is acceptable regarding that not only labels but the structure itself differs in JSON format.